### PR TITLE
(feat taxprofiler): Add store command

### PR DESCRIPTION
**Pull request by sofstam**
### Added

- `cg workflow taxprofiler store case_id`

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b store -a
    ```

### How to test

- [x] `cg workflow taxprofiler store richurchin`
Screenshot 2024-03-18 at 17 02 50

- [x] `cg workflow rnafusion store nearbyjoey`

Screenshot 2024-03-18 at 17 06 26

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [x] Tests executed by @sofstam 
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/cli/workflow/nf_analysis.py [hunk 1]
```diff
@@ -193,3 +193,35 @@ def store_housekeeper(context: CGConfig, case_id: str, dry_run: bool) -> None:
...
         LOG.error(f"Could not store bundle in Housekeeper and StatusDB: {error}!")
         raise click.Abort()
+
+
+@click.command("store")
+@ARGUMENT_CASE_ID
+@OPTION_DRY
+@click.pass_context
+def store(context: click.Context, case_id: str, dry_run: bool) -> None:
+    """Generate deliverables files for a case and store in Housekeeper if they
```
Comment (ivadym) [inline cg/cli/workflow/nf_analysis.py:203]:
```suggestion
    """Generate deliverable files for a case and store in Housekeeper if they
```

File: tests/cli/workflow/taxprofiler/test_cli_taxprofiler_compound_commands.py [hunk 2]
```diff
@@ -3,7 +3,7 @@
 from _pytest.logging import LogCaptureFixture
 from click.testing import CliRunner
 
-from cg.cli.workflow.taxprofiler.base import start, start_available, taxprofiler
+from cg.cli.workflow.taxprofiler.base import start, start_available, taxprofiler, store
```
Comment (ivadym) [inline tests/cli/workflow/taxprofiler/test_cli_taxprofiler_compound_commands.py:6]: store and CgError are not used

We have some pre-commit hooks to check for these kind of things FYI: https://atlas.scilifelab.se/infrastructure/tech_stack/editors/pycharm/#install-pre-commit

File: tests/cli/workflow/nf_analysis/test_cli_compound_commands.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+from click.testing import CliRunner
+
+from cg.apps.hermes.hermes_api import HermesApi
+from cg.apps.hermes.models import CGDeliverables
+from cg.apps.housekeeper.hk import HousekeeperAPI
+from cg.cli.workflow.base import workflow as workflow_cli
+from cg.constants import EXIT_SUCCESS, Workflow
+from cg.meta.workflow.nf_analysis import NfAnalysisAPI
+from cg.models.cg_config import CGConfig
+from tests.cli.workflow.conftest import mock_analysis_flow_cell
```
Comment (ivadym) [inline tests/cli/workflow/nf_analysis/test_cli_compound_commands.py:15]:
```suggestion
```

**Review by ivadym — APPROVED** [review]
Nice, looks good ⭐

File: tests/cli/workflow/taxprofiler/test_cli_taxprofiler_compound_commands.py (hunk 2 above)
Comment (sofstam) [reply tests/cli/workflow/taxprofiler/test_cli_taxprofiler_compound_commands.py:6]: Thanks, I was not aware of this one :)

File: cg/cli/workflow/nf_analysis.py [hunk 4]
```diff
@@ -193,3 +193,35 @@ def store_housekeeper(context: CGConfig, case_id: str, dry_run: bool) -> None:
...
+    )
+    if not is_latest_analysis_qc and not analysis_api.trailblazer_api.is_latest_analysis_completed(
+        case_id=case_id
+    ):
+        LOG.error(
+            "Case not stored. Trailblazer status must be either QC or COMPLETE to be able to store"
+        )
+        return
+
+    # Avoid storing a case without QC checks previously performed
```
Comment (henrikstranneheim) [inline cg/cli/workflow/nf_analysis.py:218]:
```suggestion
```

File: tests/cli/workflow/nf_analysis/test_cli_compound_commands.py [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+    request: FixtureRequest,
+    mocker,
+):
+    """Test store command fails when a case did not finish for a workflow."""
+    caplog.set_level(logging.INFO)
+
+    # GIVEN each fixture is being initialised
+    context: CGConfig = request.getfixturevalue(f"{workflow}_context")
+
+    # GIVEN CASE ID where analysis finish is not mocked
```
Comment (henrikstranneheim) [inline tests/cli/workflow/nf_analysis/test_cli_compound_commands.py:94]:
```suggestion
    # GIVEN a case id where analysis finish is not mocked
```

File: tests/cli/workflow/nf_analysis/test_cli_compound_commands.py [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+    context.meta_apis["analysis_api"].housekeeper_api = real_housekeeper_api
+
+    # Make sure the bundle was not present in hk
+    assert not context.housekeeper_api.bundle(case_id_fail)
+
+    # WHEN running command
+    result_fail = cli_runner.invoke(workflow_cli, [workflow, "store", case_id_fail], obj=context)
+
+    # THEN bundle exist status should be
+    assert result_fail.exit_code != EXIT_SUCCESS
```
Comment (henrikstranneheim) [inline tests/cli/workflow/nf_analysis/test_cli_compound_commands.py:115]: WWould be nice to assert that the bundle was created.

**Review by henrikstranneheim — APPROVED** [review]
Great that you take the time to generalize and improve the code as you go along. 👍

File: tests/cli/workflow/nf_analysis/test_cli_compound_commands.py (hunk 6 above)
Comment (sofstam) [reply tests/cli/workflow/nf_analysis/test_cli_compound_commands.py:120]: @henrikstranneheim Updated the tests.

File: cg/cli/workflow/nf_analysis.py [hunk 7]
```diff
@@ -193,3 +193,34 @@ def store_housekeeper(context: CGConfig, case_id: str, dry_run: bool) -> None:
...
+
+    if (
+        is_latest_analysis_qc
+        or not analysis_api.get_metrics_deliverables_path(case_id=case_id).exists()
+    ):
+        LOG.info(f"Generating metrics file and performing QC checks for {case_id}")
+        context.invoke(metrics_deliver, case_id=case_id, dry_run=dry_run)
+    LOG.info(f"Storing analysis for {case_id}")
+    context.invoke(report_deliver, case_id=case_id, dry_run=dry_run)
+    context.invoke(store_housekeeper, case_id=case_id, dry_run=dry_run)
```
Comment (ChrOertlin) [inline cg/cli/workflow/nf_analysis.py:226]:
```suggestion
    analysis.api.report_deliver()
    analysis_api.store_housekeeper()
```

Following the `invoke`it looks like the analysis_api  that is initiated here has these functions that are invoked via the CLI. Why do we not call them directly here?

Comment (ChrOertlin) [reply cg/cli/workflow/nf_analysis.py:226]: Seems to also be the case for the `metrics_deliver`

Comment (ChrOertlin) [inline cg/cli/workflow/nf_analysis.py:226]: ALso I think all this logic can be done within the analysis api and then have a wrapper that is called here that does all this logic.

**Review by ChrOertlin — APPROVED** [review]
Nice!

**Comment by sofstam** [comment]
Deploy stage:
Screenshot 2024-03-21 at 11 13 54

**Comment by sofstam** [comment]
Deploy production
Screenshot 2024-03-21 at 11 18 15

**Pull request merged**